code/ship/ship.cpp

```diff
@@ -13396,6 +13395,29 @@ int ship_get_subsys_index(ship *sp, const char* ss_name, int error_bypass)
 	return -1;
 }
 
+/**
+* Returns the index number of the ship_subsys parameter
+*/
+int ship_get_subsys_index(ship *shipp, ship_subsys *subsys, int error_bypass)
+{
+	int count;
+	ship_subsys *ss;
+
+	count = 0;
+	ss = GET_FIRST(&shipp->subsys_list);
+	while (ss != END_OF_LIST(&shipp->subsys_list)) {
+		if (ss == subsys)
+			return count;
+		count++;
+		ss = GET_NEXT(ss);
+	}
+
+	if (!error_bypass)
+		Int3();
```

Comment:
`Int3()` is more or less deprecated since it doesn't show an error message. This should be replaced by an `Assertion` with a proper error message.

Reply:
I was mirroring the other functions that performed similar tasks.  In my view all of them should have the Int3() removed since that was probably only needed during development. :man_shrugging: 

The best thing IMO would be to make another PR to remove those Int3()s.  I'll do that.

Reply:
The other functions could be changed (in another PR) but that doesn't mean that new code should use old functions just to keep the style of existing functions.